features/networking/ovn.feature

```diff
@@ -0,0 +1,147 @@
+      | object_type       | ds                       |
+      | object_name_or_id | ovnkube-master           |
+      | n                 | openshift-ovn-kubernetes |
+    Then the step should succeed
+    Given I have a project
+    And I obtain test data file "networking/pod-for-ping.json"
+    When I run the :create client command with:
+      | f | pod-for-ping.json |
+    Then the step should succeed
+    #Now scale down CNO pod to 1 and check whether hello-pod is synced to NB db
+    Given I run the :scale admin command with:
+      | resource | deployment       |
+      | name     | network-operator |
+      | replicas | 1                |
+    Then the step should succeed
+    #A minimum wait for 30 seconds is tested to reflect CNO deployment to be effective which will then re-spawn ovn pods
+    Given 30 seconds have passed
+    And admin waits for all pods in the "openshift-ovn-kubernetes" project to become ready up to 60 seconds
+    #Checking whether Kube API data is synced on OVN NB db which in this case is a test-pod created in earlier steps
+    Given I store the ovnkube-master "north" leader pod in the clipboard
+    And admin executes on the pod:
+      | bash | -c | ovn-nbctl list logical_switch_port |
+    Then the step should succeed
+    And the output should contain:
+      | hello-pod |
+  
+  # @author anusaxen@redhat.com
+  # @case_id OCP-30057
+  @admin
+  @destructive
+  Scenario: OVN DB should be updated correctly if a resource only exist in NB db but not in Kube API	
+    Given the env is using "OVNKubernetes" networkType
+    Given I have a project
+    And evaluation of `project.name` is stored in the :hello_pod_project clipboard
+    And I have a pod-for-ping in the project
+    #Checking whether Kube API data is synced on OVN NB db which in this case is a hello-pod created in earlier steps
+    Given I store the ovnkube-master "north" leader pod in the clipboard
+    And admin executes on the pod:
+      | bash | -c | ovn-nbctl list logical_switch_port |
+    Then the step should succeed
+    And the output should contain:
+      | hello-pod |
+    # Now scale down CNO pod to 0
+    Given admin uses the "openshift-network-operator" project
+    And I run the :scale admin command with:
+      | resource | deployment       |
+      | name     | network-operator |
+      | replicas | 0                |
+    Then the step should succeed
+    Given I register clean-up steps:
+    """
+    Given admin uses the "openshift-network-operator" project
+    And I run the :scale admin command with:
+      | resource | deployment       |
+      | name     | network-operator |
+      | replicas | 1                |
+    Then the step should succeed
+    """
+    When I run the :delete admin command with:
+      | object_type       | ds                       |
+      | object_name_or_id | ovnkube-master           |
+      | n                 | openshift-ovn-kubernetes |
+    Then the step should succeed
```

Comment:
same question, should we ensure the pods are terminated?

Reply:
@rbbratta IMO, Kube API response should be enough to make sure that. The positive response from KUBE API will let 129 to pass.Pods not terminated even if ds is deleted would be a deployment team test/bug they already must be running similar checks.

Reply:
I though I saw the delete ds return before and then the pods in Terminating state, so I think it might take a few seconds before the pods finish terminating.  It might not be an issue, but I think there could be race.

Reply:
@rbbratta No worries. Added a step to make sure that. had to modify ```existing pods die with labels:``` step a bit to not break existing scenarios and support admin capabilities as well for accessing system projects.

Reply:
@anuragthehatter you can replace the call with the step `Given admin ensure...`

Reply:
Ah right. Fixed it.Thanks and re-ran to make sure